distribution/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM centos:7
+
+# Project version defined in pom.xml is passed as an argument
+ARG VERSION
+
+# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
+RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+
+# The drill tarball is generated upon building the Drill project
+COPY target/apache-drill-$VERSION.tar.gz /tmp
+
+# Drill binaries are extracted into the '/opt/drill' directory
+RUN mkdir /opt/drill
+RUN tar -xvzf /tmp/apache-drill-$VERSION.tar.gz --directory=/opt/drill --strip-components 1
```

Comment:
A handy trick I learned is to do the container setup in a script, say `setup.sh` that is copied into the container and run. This avoids creating a bunch of extra Docker layers as the set of steps gets larger.

Reply:
I'm actually in favor doing setup within the Dockerfile, not in a `setup.sh`. The layers are actually advantageous because if you change a step at the end of the Dockerfile or add a step at the end, only the diff is rebuilt. All the layers before that new step can be reused from the docker's cache. Furthermore it speeds up deploying updated docker containers since only the diff has to be downloaded. If you use a master `setup.sh` script there will be essentially only one layer, so even small changes to the setup script will require a complete rebuild of your container, and then a complete redownload by clients.

The trick is that you want the big infrequently changing setup tasks to be toward the beginning of your docker file to take full advantage of layer caching. And you want the small lightweight setup tasks to be towards the end of the Dockerfile, so that if you change them hardly anything needs to be rebuilt, uploaded, and downloaded.

Reply:
@paul-rogers My initial implementation did have a init script - i removed it for this PR - to make it as simple as possible. I'll need to include it later for distributed mode support, if I cannot figure out a different approach for a keep alive probe. 

@ilooner , agree with regard to layers - i find it very useful. I don't intend to move the tar download to setup.sh - just the config and start-up parts (eventually, in a new PR).